libcalico-go/lib/ipam/ipam.go

```diff
@@ -1993,12 +2004,15 @@ func (c ipamClient) convertBackendToIPAMConfig(cfg *model.IPAMConfig) *IPAMConfi
 func (c ipamClient) ensureConsistentAffinity(ctx context.Context, b *model.AllocationBlock) error {
 	// Retrieve node for this allocation. We do this so we can clean up affinity for blocks
 	// which should no longer be affine to this host.
-	host := getHostAffinity(b)
-	logCtx := log.WithFields(log.Fields{"cidr": b.CIDR, "host": host})
+	affinityCfg := getHostAffinity(b)
```

Comment:
Seems like you opted to keep this function scoped only to `host:` type affinities, which I think makes any functions that call it (`ensureConsistentAffinity` and `ReleaseBlockAffinity`) silently fail to work for virtual affinities. 

I think I would rather we:
- Update `getHostAffinity` to handle virtual affinities as well.
- Have `ensureConsistentAffinity` and `ReleaseBlockAffinity` explicitly decide whether they support virtual affinities by checking the `Type` on the returned `AffinityConfig` and taking action based on that. 

It will make this code easier to follow.

Reply:
Worth calling out - I think it is correct to skip this function for virtual affinities (probably with a debug log as well).